### PR TITLE
PIMOB-1401: Error handling was added.

### PR DESCRIPTION
**Pull request by chintan-soni-cko**
## Issue

[PIMOB-1401](https://checkout.atlassian.net/browse/PIMOB-1401)

## Proposed changes

Error handling was implemented for expiry date.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Reviewers assigned
* [X] I have performed a self-review of my code and manual testing
* [X] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if applicable)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you choose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt [hunk 1]
```diff
@@ -6,4 +6,22 @@ internal data class ExpiryDateComponentState(
     val inputState: InputComponentState
 ) {
     val expiryDate = inputState.inputFieldState.text
+
+    fun hideError() = switchErrorVisibility(false)
+
+    fun showError(errorMessageId: Int) = with(inputState.errorState) {
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt:12]: Please can you add unit test

File: frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt [hunk 2]
```diff
@@ -6,4 +6,22 @@ internal data class ExpiryDateComponentState(
...
+        textId.value = errorMessageId
+        switchErrorVisibility(true)
+    }
+
+    fun showError(errorMessage: String) = with(inputState.errorState) {
```
Comment (deepesh-vasthimal-cko) [inline frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt:17]: Please can you add unit test

File: frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt (hunk 1 above)
Comment (chintan-soni-cko) [reply frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt:12]: @deepesh-vasthimal-cko  I have divided tasks into smaller PR. So, Unit tests for this component would be separate PR. You can find this ticket for the unit test. [/PIMOB-1404](https://checkout.atlassian.net/browse/PIMOB-1404)

File: frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt (hunk 2 above)
Comment (chintan-soni-cko) [reply frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateComponentState.kt:17]: https://github.com/checkout/frames-android/pull/116#discussion_r959295401

**Review by maxim-nosov-cko — APPROVED** [review]

**Pull request merged**